lib/compress/zstd_compress.c

```diff
@@ -7172,20 +7390,78 @@ size_t ZSTD_convertBlockSequences(ZSTD_CCtx* cctx,
     return 0;
 }
 
-typedef struct {
-    size_t nbSequences;
-    size_t blockSize;
-    size_t litSize;
-} BlockSummary;
+#if defined(__AVX2__)
+
+/* C90-compatible alignment macro (GCC/Clang). Adjust for other compilers if needed. */
+#if defined(__GNUC__)
+#  define ALIGNED32 __attribute__((aligned(32)))
+#elif defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L) /* C11 */
+#  define ALIGNED32 alignas(32)
+#else
+   /* this compiler will require its own alignment instruction */
+#  define ALIGNED32
+#endif
```

Comment:
I know there are other places we align. Should we unify this into a common macro in `compilers.h`?